universal-application-tool-0.0.1/app/repository/QuestionRepository.java

```diff
@@ -33,6 +33,52 @@ public QuestionRepository(EbeanConfig ebeanConfig, DatabaseExecutionContext exec
         executionContext);
   }
 
+  static class PathConflictDetector {
+    private Optional<Question> conflictedQuestion = Optional.empty();
+    private final String newPath;
+
+    PathConflictDetector(String newPath) {
+      this.newPath = checkNotNull(newPath);
+    }
+
+    public Optional<Question> getConflictedQuestion() {
+      return conflictedQuestion;
+    }
+
+    public boolean checkConflict(Question question) {
+      boolean proceed = true;
+      if (pathConflicts(question.getPath(), newPath)) {
+        conflictedQuestion = Optional.of(question);
+        proceed = false;
+      }
+      return proceed;
+    }
+
+    public static boolean pathConflicts(String path, String otherPath) {
```

Comment:
nit: this can be a package-private since the test is in the same package

Reply:
Done.